packages/liquidator/src/liquidator.js

```diff
@@ -269,23 +271,22 @@ class Liquidator {
 
       // Note: query the time again during each iteration to ensure the deadline is set reasonably.
       const currentBlockTime = this.financialContractClient.getLastUpdateTime();
-      const syntheticTokenBalance = this.toBN(await this.syntheticToken.methods.balanceOf(this.account).call());
+
+      // TODO: right now the `processPosition` method takes in maxTokensToLiquidateWei and syntheticTokenBalance. These
```

Comment:
Don't you think it could be useful to distinguish between the bot's current balance and the amount of balance its allocating to liquidate with? I.e. it allows the caller to set a limit on how many tokens to liquidate

Reply:
ye, this is something I spent quite a lot of time thinking about. Previously, this parameter was simply the bots token balance. Within the `liquidatePositions` method it also read in the current token balance. Then, the WDF logic took in two parameters that were _the same_.

What I did here is implement this to be the _same_ as the previous implementation with the goal of not modifying key logic that we know to work within the WDF. In a separate PR we should refactor this. There were a few other things in how the WDF is called and the data it returns that I'd also like to change.

I felt it best to mimic the current logic as much as posible and leave these changes for later. do you agree?

Reply:
Yes agreed, minimize churn for this PR.